irohad/main/application.cpp

```diff
@@ -766,6 +739,31 @@ Irohad::RunResult Irohad::initConsensusGate() {
 
   auto &block =
       boost::get<expected::ValueOf<decltype(block_var)>>(&block_var)->value;
+
+  // reject_delay and local_counter are local mutable variables of lambda
+  auto delay = [max_rounds_delay = max_rounds_delay_,
+                reject_delay = std::chrono::milliseconds(0),
+                local_counter = static_cast<uint64_t const>(0),
+                kMaxLocalCounter = static_cast<uint64_t const>(2)](
+                   ConsensusOutcomeType type) mutable {
+    const auto kMaxDelay(max_rounds_delay);
+    const auto kMaxDelayIncrement(std::chrono::milliseconds(1000));
+    if (type == ConsensusOutcomeType::kReject
+        or type == ConsensusOutcomeType::kNothing) {
+      // Increment reject_counter each local_counter calls of function
+      ++local_counter;
+      if (local_counter == kMaxLocalCounter) {
+        local_counter = 0;
+        if (reject_delay < kMaxDelay) {
+          reject_delay += std::min(kMaxDelay, kMaxDelayIncrement);
+        }
+      }
+    } else {
+      reject_delay = std::chrono::milliseconds(0);
+    }
+    return reject_delay;
+  };
```

Comment:
if we move this to const variable in defined in ` irohad/main/impl/consensus_init.cpp` and declared in corresp. header, together with zero-delay func, we can decrease the size of `application.o`.